zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/handlers/ListViewProcessInstanceFromProcessInstanceHandlerTest.java

```diff
@@ -0,0 +1,224 @@
+            .filter(intent -> !intents2Handle.contains(intent))
+            .collect(Collectors.toSet());
+
+    intents2Ignore.stream()
+        .forEach(
+            intent -> {
+              final Record<ProcessInstanceRecordValue> processInstanceRecord = createRecord(intent);
+              // when - then
+              assertThat(underTest.handlesRecord(processInstanceRecord))
+                  .as("Does not handle intent %s", intent)
+                  .isFalse();
+            });
+  }
+
+  @Test
+  public void shouldGenerateIds() {
+    // given
+    final long expectedId = 123;
+    final ProcessInstanceRecordValue processInstanceRecordValue =
+        ImmutableProcessInstanceRecordValue.builder()
+            .from(factory.generateObject(DecisionRecordValue.class))
+            .withBpmnElementType(BpmnElementType.PROCESS)
+            .withProcessInstanceKey(expectedId)
+            .build();
+
+    final Record<ProcessInstanceRecordValue> processInstanceRecord =
+        factory.generateRecord(
+            ValueType.PROCESS_INSTANCE,
+            r ->
+                r.withIntent(ProcessInstanceIntent.ELEMENT_ACTIVATING)
+                    .withValue(processInstanceRecordValue));
+
+    // when
+    final var idList = underTest.generateIds(processInstanceRecord);
+
+    // then
+    assertThat(idList).containsExactly(String.valueOf(expectedId));
+  }
+
+  @Test
+  public void shouldCreateNewEntity() {
+    // when
+    final var result = (underTest.createNewEntity("id"));
+
+    // then
+    assertThat(result).isNotNull();
+    assertThat(result.getId()).isEqualTo("id");
+  }
+
+  @Test
+  void shouldAddEntityOnFlush() {
+    // given
+    final ProcessInstanceForListViewEntity inputEntity =
+        new ProcessInstanceForListViewEntity()
+            .setId("111")
+            .setProcessName("process")
+            .setProcessVersion(2)
+            .setProcessDefinitionKey(444L)
+            .setBpmnProcessId("bpmnProcessId")
+            .setPosition(123L);
+    final BatchRequest mockRequest = mock(BatchRequest.class);
+
+    final Map<String, Object> expectedUpdateFields = new LinkedHashMap<>();
+    expectedUpdateFields.put(ListViewTemplate.PROCESS_NAME, "process");
+    expectedUpdateFields.put(ListViewTemplate.PROCESS_VERSION, 2);
+    expectedUpdateFields.put(ListViewTemplate.PROCESS_KEY, 444L);
+    expectedUpdateFields.put(ListViewTemplate.BPMN_PROCESS_ID, "bpmnProcessId");
+    expectedUpdateFields.put(POSITION, 123L);
+
+    // when
+    underTest.flush(inputEntity, mockRequest);
+
+    // then
+    verify(mockRequest, times(1)).upsert(indexName, "111", inputEntity, expectedUpdateFields);
+  }
+
+  @Test
+  void shouldUpdateEntityFromRecord() {
+    // given
+    final ProcessInstanceRecordValue processInstanceRecordValue =
+        ImmutableProcessInstanceRecordValue.builder()
+            .from(factory.generateObject(ProcessInstanceRecordValue.class))
+            .withProcessInstanceKey(66L)
```

Comment:
Any reason not to use the random generated values from `factory` and also do the assertions on it's values instead of static typed values?